### PR TITLE
BIP143 clarifying semantics of ACP|SINGLE

### PR DESCRIPTION
**Pull request by jl2012**
This PR clarifies the semantics change when ANYONECANPAY is used with SINGLE, with other minor clarifications

### CODE REVIEW

**Comment by jl2012** [comment]
@sipa

**Comment by NicolasDorier** [comment]
ACK for me except a nit that might be a english mistake on my part:

> "does not commit --to-- the input index"

I would say

> "does not commit the input index"

**Comment by jl2012** [comment]
@NicolasDorier fixed. Thanks

File: bip-0143.mediawiki [hunk 1]
```diff
@@ -1,4 +1,4 @@
-﻿<pre>
+<pre>
```
Comment (luke-jr) [inline bip-0143.mediawiki]: ...

Comment (btcdrak) [inline bip-0143.mediawiki]: I dont see it?

Comment (jl2012) [inline bip-0143.mediawiki]: @luke-jr I think there was a hidden character but this PR removes it. Isn't it?

File: bip-0143.mediawiki [hunk 2]
```diff
@@ -28,37 +28,41 @@ A new transaction digest algorithm is defined, but only applicable to sigops in
...
      6. value of the output spent by this input (8-byte little endian)
      7. nSequence of the input (4-byte little endian)
      8. hashOutputs (32-byte hash)
      9. nLocktime of the transaction (4-byte little endian)
     10. sighash type of the signature (4-byte little endian)
 
-All components in the original algorithm, including the behavior <code>OP_CODESEPERATOR</code>, remains unchanged. The only difference is the way of serialization and the inclusion of amount being spent.
+Semantics of the original sighash types remain unchanged, except the followings:
+# The way of serialization is changed;
+# All sighash types commit the amount being spent by the signed input;
```
Comment (sipa) [inline bip-0143.mediawiki]: commit *to* the amount being spent

File: bip-0143.mediawiki (hunk 1 above)
Comment (MarcoFalke) [inline bip-0143.mediawiki]: https://en.wikipedia.org/wiki/Byte_order_mark#UTF-8

Comment (jl2012) [inline bip-0143.mediawiki]: @MarcoFalke I think there was a BOM before, not now

File: bip-0143.mediawiki (hunk 2 above)
Comment (jl2012) [inline bip-0143.mediawiki]: revised and squashed

**Pull request merged**